drivers/at24cxxx/at24cxxx.c

```diff
@@ -0,0 +1,248 @@
+/**
+ * @ingroup     drivers_at24cxxx
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the AT24C128 / AT24C256 EEPROM units.
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <errno.h>
+#include <string.h>
+
+#define ENABLE_DEBUG  (0)
+#include "debug.h"
+#include "assert.h"
+#include "xtimer.h"
+#include "byteorder.h"
+
+#include "include/at24cxxx_params.h"
+#include "include/at24cxxx_defines.h"
+#include "at24cxxx.h"
+
+static int _read(const at24cxxx_t *dev, uint32_t pos, uint8_t *data, size_t len)
+{
+    int check;
+    uint8_t polls = AT24CXXX_MAX_POLLS;
+
+    while ((check =
+                i2c_read_regs(dev->params.i2c, dev->params.dev_addr, pos, data,
+                              len,
+                              I2C_REG16)) == -ENXIO && --polls) {
+        xtimer_usleep(AT24CXXX_POLL_DELAY_MS * US_PER_MS);
+    }
+    DEBUG("_read::i2c_read_regs check: %d; polls: %d\n", check, polls);
+    return check < 0 ? check : (int)len;
+}
+
+static int _write(const at24cxxx_t *dev, uint32_t pos, const uint8_t *data,
+                  size_t len)
+{
+    int check = 0;
+    size_t wr = 0;
+    size_t chunk_len;
+
+    while (!check && (chunk_len = len - wr)) {
+        const uint8_t *chunk_data = data + wr;
+        size_t max_chunk_len =
+            (AT24CXXX_PAGE_SIZE - (pos % AT24CXXX_PAGE_SIZE));
+        chunk_len = chunk_len < max_chunk_len ? chunk_len : max_chunk_len;
```

Comment:
how about
`chunk_len = min(len, AT24CXXX_PAGE_SIZE - (pos & (AT24CXXX_PAGE_SIZE - 1)));`

Reply:
Ok, gonna change that